docs/client-side/using-gulp.rst

```diff
@@ -0,0 +1,219 @@
+	  "name": "MyWebApp",
+	  "private": true,
+	  "devDependencies": {
+	    "gulp": "3.9.0"
+	  }
+	}
+
+The packages are identified using the numbering scheme **<major>.<minor>.<patch>** according to the `SemVer <http://semver.org/>`_ (semantic versioning) standard.
+
+After typing the name of the package and a colon, Intellisense shows a simplified list of versions with the most common choices:  
+
+.. image:: using-gulp/_static/semver.png
+
+-  The top item in the Intellisense list is considered the latest stable version of the package. 
+-  The carat ^ symbol matches the most recent major version.
+-  The tilde ~ matches the most recent minor version. 
+
+Add the Gulp config file
+------------------------
+1.	In Solution Explorer, right-click the project.
+
+#.	Select Add > New Item.
+
+#.	Select **Gulp Configuration file**.
+
+#.	Leave the default name, "gulpfile.js".
+
+#.	Click **Add**.
+
+#.  Right-click Dependencies > NPM and select **Restore Packages** from the context menu.
+
+Define a "Hello World" task
+---------------------------
+Open gulpfile.js and add the following line.
+
+.. code-block:: javascript
+   :emphasize-lines: 4
+
+	var gulp = require('gulp');
+
+	gulp.task('default', function () {
+	    console.log('hello, world');
+	});
+
+In Solution Explorer, right-click gulpfile.js and select Task Runner Explorer. This opens the Task Runner Explorer window.
+Task Runner Explorer shows the list of Gulp tasks. So far, we've only defined one ('default').
+ 
+.. image:: using-gulp/_static/task-runner-1.png
+
+Right-click the default task and select **Run**.
+ 
+.. image:: using-gulp/_static/task-runner-2.png
+
+You will see the output from the task. In this case, the task just writes "hello, world" to the console.
+
+.. image:: using-gulp/_static/task-runner-3.png
+
+Use Gulp to copy files
+----------------------
+Here is an example of a more useful task.
+
+.. code-block:: javascript
+
+	var gulp = require('gulp');
+
+	var paths = {
+	    src: "./Assets/**/*.js",
+	    dest: "./wwwroot/js/"
+	}
+
+	gulp.task('default', function () {
+	    return gulp.src(paths.src)         // Returns a stream
+	        .pipe(gulp.dest(paths.dest))   // Pipes the stream somewhere
+	});
+
+This task copies JS files from an *Assets* folder into *wwwroot/js*.
+ 
+-	The gulp.src method returns a stream of files. You can use `file globbing <https://github.com/isaacs/node-glob>`_ to match multiple files. In this case, we are matching every .js file under Assets. 
+-	The gulp.dest method writes the streamed files to a destination folder.
+-	The pipe method pipes the files from src to dest.
+
+The real power of Gulp is that you can pipe a file through multiple plugins. The output from each stage becomes the input to the next. For example, your pipeline might have these stages:
+
+compile TypeScript > run JSHint > minify 
+
+Use Gulp to run JSHint
+----------------------
+Let's add JSHint to our pipeline. We'll also add a "clean" task.
```

Comment:
Maybe add a transition sentence that says, to do real work with Gulp, you'll use plugins. Link to http://gulpjs.com/plugins/